iroh-net/src/magicsock/node_map/node_state.rs

```diff
@@ -1310,6 +1361,27 @@ impl RemoteInfo {
     pub fn has_send_address(&self) -> bool {
         self.relay_url.is_some() || !self.addrs.is_empty()
     }
+
+    /// Returns a deduplicated list of [`Source`]s merged from all address in the [`RemoteInfo`].
+    ///
+    /// Deduplication is on the (`Source`, `Duration`) tuple, so you will get multiple [`Source`]s
+    /// for each `Source` variant, if those addresses were discovered from the same [`Source`]
+    /// at different times.
+    ///
+    /// The list is sorted from least to most recent [`Source`].
```

Comment:
It is sorted only by duration and not by source at all?  Or am I reading that wrong?

Reply:
yes, only by duration.